vinyldns/recordsets.go

```diff
@@ -167,6 +167,41 @@ func (c *Client) RecordSetDelete(zoneID, recordSetID string) (*RecordSetUpdateRe
 	return resource, nil
 }
 
+// RecordSetChanges retrieves the RecordSetChanges response for the Zone and ListFilter it's passed.
+func (c *Client) RecordSetChanges(zoneID string, f ListFilter) (*RecordSetChanges, error) {
+	rsc := &RecordSetChanges{}
+	err := resourceRequest(c, recordSetChangesEP(c, zoneID, f), "GET", nil, rsc)
+	if err != nil {
+		return &RecordSetChanges{}, err
+	}
+
+	return rsc, nil
+}
+
+// RecordSetChangesListAll retrieves the complete list of record set changes for the Zone ListFilter criteria passed.
+// Handles paging through results on the user's behalf.
+func (c *Client) RecordSetChangesListAll(zoneID string, filter ListFilter) ([]RecordSetChange, error) {
+	if filter.MaxItems > 100 {
+		return nil, fmt.Errorf("MaxItems must be between 1 and 100")
+	}
+
+	rsc := []RecordSetChange{}
+
+	for {
+		resp, err := c.RecordSetChanges(zoneID, filter)
+		if err != nil {
+			return nil, err
+		}
+
+		rsc = append(rsc, resp.RecordSetChanges...)
+		filter.StartFrom = resp.NextID
+
+		if len(filter.StartFrom) == 0 {
```

Comment:
unclear if the `NextID` will be a number or 0 here.  The `StartFrom` / `NextID` should be `null` if there is no next page.  Just looking at the code API side, and can confirm NextID will be a `String`...

```scala
case class ListRecordSetChangesResponse(
    zoneId: String,
    recordSetChanges: List[RecordSetChangeInfo] = Nil,
    nextId: Option[String],
    startFrom: Option[String],
    maxItems: Int)
```

Reply:
@pauljamescleary Note line 197, where we set `filter.StartFrom` to the value of `resp.NextID` from the last recordsetchanges response. Now note how recordsetchanges `go-vinyldns` JSON unmarshal-ing  works; we convert the VinylDNS JSON to the following struct:

1. `NextID` is a [string, not a number](https://github.com/vinyldns/go-vinyldns/blob/recordsetchanges/vinyldns/recordsets_resources.go#L32)
2. If the JSON has no `nextId`, then `resp.NextID` will be `""` and have no value. Here we are checking its length to confirm it is in fact empty: `if len(filter.StartFrom) == 0 {`